src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -282,8 +284,33 @@ private static Credentials secretToCredentials(Secret secret) {
                     data.get(OPENSHIFT_SECRETS_DATA_USERNAME),
                     data.get(OPENSHIFT_SECRETS_DATA_SSHPRIVATEKEY));
         default:
-            logger.log(Level.WARNING,
-                    "Unknown secret type: " + secret.getType());
+			String type = secret.getType();
+			if (type == null || type.length() == 0) {
+				// the type field is marked optional in k8s.io/api/core/v1/types.go,
+				// default to OPENSHIFT_SECRETS_DATA_SECRET_TEXT in this case
+				String text = "";
+				if (data != null && data.size() > 0) {
+					// convert to JSON for parsing ease in pipelines
+					try {
+						text = new ObjectMapper().writeValueAsString(data);
+					} catch (JsonProcessingException e) {
+					}
+				} else {
+					data = secret.getStringData();
+					if (data != null && data.size() > 0) {
+						// convert to JSON for parsing ease in pipelines
+						try {
+							text = new ObjectMapper().writeValueAsString(data);
+						} catch (JsonProcessingException e) {
+						}
+					}
+				}
+				if (text.length() > 0) {
+					return newSecretTextCredential(secretName, text);
+				}
+			}
+			logger.log(Level.WARNING,
+					"Unknown secret type: " + secret.getType() + " " + secret.getData() + " " + secret.getStringData());
```

Comment:
dumping the secret data to the log is probably a bad idea.

Reply:
duh :-/ ... removed